Improve the LTM wording (#246)
apache/accumulo-website: contributor/versioning.md

1. Periodically release a new LTM `major.minor.0` version (approximately every 2 years),
2. Maintain the LTM releases with `major.minor.(patch++)` releases until 1 year after the next LTM,
3. Support upgrades from both the immediately preceding LTM and non-LTM versions

This strategy implies that no more than one or two LTM releases will be

> actively maintained at any given time, with a one-year overlap.

Reviewer: I don't think this is more correct. The hyphen implies this phrase can be treated as a single word, as in an `overlap of type 'one-year'`. I don't think that makes sense grammatically.

If the existing wording isn't clear, we could change it to:
```suggestion
actively maintained at any given time, with one year of overlap.
```

> Non-LTM versions:

Reviewer: This is a clear Non-LTM section, but there is no corresponding clear LTM section above the earlier list of items. Also, should these headers have `###` section prefixes?

Author: I updated the sections using ```###``` and replaced the semantic versioning link on the front page to link to this versioning page.  I thought this page is more useful and it contains link to the semantic versioning documentation.

1. Release intermediate non-LTM `major.minor.0` versions that are not expected to receive patch/bugfix releases,

> 2. Roll patches/bugfixes targeting non-LTM versions into the next `major.minor.0` release

Reviewer: I'm not sure how I feel about them being split up like this. The idea was to describe the entire strategy in one concise list. Spreading it out makes it seem like there are two strategies. It's probably fine separate... I'm just not sure that it's better or not.

Author: Yeah I wasn't sure either.  My thinking was that there are probably some managers or non technical folks that do not care about development non LTM releases and I was trying to have a part just for their attention span.

Goals of the LTM strategy:
- Streamline maintaining multiple versions of Accumulo
- Encourage development on newer versions
- Provide greater confidence for users to upgrade and receive updates (within one year)

> - Help reduce the amount of work done by limiting upgrade paths

Reviewer: :+1:  bullet points are way better than prose.

Note: the above strategy is a declaration of **intent** only. We use the term
"Long Term Maintenance" rather than the more familiar "Long Term Support" or

> "Long Term Stable" terms to avoid confusion over the implication of warranties

Reviewer: Yours is better. But you might be able to make it even shorter:

```suggestion
"Long Term Stable" terms to avoid any implication of warranties
```
